dashboard/src/smoke_tests.py

```diff
@@ -0,0 +1,57 @@
+"""Module with class that handle smoke test results."""
+import csv
+
+
+class SmokeTests:
+    """Class that handle smoke test results."""
+
+    INPUT_FILES = {
+        "production": {
+            "logs": "smoketests_prod.log",
+            "results": "smoketests_prod.results"
+        },
+        "stage": {
+            "logs": "smoketests_stage.log",
+            "results": "smoketests_stage.results"
+        }
+    }
+
+    def __init__(self):
+        """Construct an instance of the class."""
+        self._results = {}
+        self._logs = {}
+
+    def read_logs(self):
+        """Read logs for all systems (stage, production)."""
+        for system, filenames in SmokeTests.INPUT_FILES.items():
+            input_file = filenames["logs"]
+            with open(input_file) as fin:
+                self._logs[system] = fin.read()
+
+    def read_results(self):
+        """Read results generated by all smoke tests."""
+        for system, filenames in SmokeTests.INPUT_FILES.items():
+            input_file = filenames["results"]
+            with open(input_file) as fin:
+                self._results[system] = fin.read().strip() == "0"
+
+    @property
+    def results(self):
+        """Getter for the 'results' attribute."""
+        return self._results
```

Comment:
One improvement that just came to my mind is that you could also read the results in this *getter* if they haven't been read before, i.e.

```python
   @property
   def results(self):
       if not self._results:
           self._results = self.read_results()
       return self._results
```

so you don't have to call the `read_results()` explicitly before accessing `results` property.

Reply:
yep, let's do this. TY!